applications/GeoMechanicsApplication/custom_constitutive/thermal_filter_law.h

```diff
@@ -0,0 +1,35 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#pragma once
+
+#include "custom_constitutive/thermal_law.h"
+
+namespace Kratos
+{
+
+class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoThermalFilterLaw : public GeoThermalLaw
+{
+public:
+    KRATOS_CLASS_POINTER_DEFINITION(GeoThermalFilterLaw);
+
+    Matrix CalculateThermalDispersionMatrix(const Properties& rProp, const ProcessInfo&) const override;
+
+private:
+    friend class Serializer;
+
+    void save(Serializer&) const override {}
+
+    void load(Serializer&) override {}
```

Comment:
Please add comments to these function bodies which explain why they are empty. For instance, "Nothing to serialize, since there are no data members, and its base class is abstract". This will resolve two code smells found by SonarQube.